mi/dataset/parser/metbk_a_dcl.py

```diff
@@ -151,5 +123,85 @@ def __init__(self,
         super(MetbkADclParser, self).__init__(config,
                                               stream_handle,
                                               exception_callback,
-                                              SENSOR_DATA_MATCHER,
-                                              METADATA_MATCHER)
+                                              '',
+                                              '')
+        self.particle_classes = None
+
+    def parse_file(self):
+        """
+        This method reads the file and parses the data within, and at
+        the end of this method self._record_buffer will be filled with all the particles in the file.
+        """
+
+        # If not set from config & no InstrumentParameterException error from constructor
+        if self.particle_classes is None:
+            self.particle_classes = (self._particle_class,)
+
+        for particle_class in self.particle_classes:
+
+            for line in self._stream_handle:
+                raw_data = line.split()
```

Comment:
this is where the regex should be used to see if there is a match for expected data, then only the data part should be used to create `raw_data`